Header header
float64 ratio
